src/plugins/multi/komga.ts

```diff
@@ -0,0 +1,272 @@
+import { fetchApi } from '@libs/fetch';
+import { Filters, FilterTypes } from '@libs/filterInputs';
+import { NovelStatus } from '@libs/novelStatus';
+import { Plugin } from '@typings/plugin';
+import { load as parseHTML } from 'cheerio';
+import { storage } from '@libs/storage';
+
+class KomgaPlugin implements Plugin.PluginBase {
+  id = 'komga';
+  name = 'Komga';
+  icon = 'src/multi/komga/icon.png';
+  version = '1.0.0';
+
+  site = storage.get('url');
+  email = storage.get('email');
+  password = storage.get('password');
+
+  async makeRequest(url: string): Promise<string> {
+    return await fetchApi(url, {
+      headers: {
+        Accept: 'application/json, text/plain, */*',
+        'Content-Type': 'application/json;charset=utf-8',
+        'Authorization': `Basic ${this.btoa(this.email + ':' + this.password)}`,
+      },
+      Referer: this.site,
+    }).then(res => res.text());
+  }
+
+  btoa(input: string = '') {
+    const chars =
+      'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/=';
+    let str = input;
+    let output = '';
+
+    for (
+      let block = 0, charCode, i = 0, map = chars;
+      str.charAt(i | 0) || ((map = '='), i % 1);
+      output += map.charAt(63 & (block >> (8 - (i % 1) * 8)))
+    ) {
+      charCode = str.charCodeAt((i += 3 / 4));
+
+      if (charCode > 0xff) {
+        throw new Error(
+          "'btoa' failed: The string to be encoded contains characters outside of the Latin1 range.",
+        );
+      }
+
+      block = (block << 8) | charCode;
+    }
+
+    return output;
+  }
+
+  flattenArray(arr: any) {
+    return arr.reduce((acc: any, obj: any) => {
+      const { children, ...rest } = obj;
+      acc.push(rest);
+
+      if (children) {
+        acc.push(...this.flattenArray(children));
+      }
+
+      return acc;
+    }, []);
+  }
```

Comment:
flat is also already a function on array, id assume hermes supports (couldent find any github posts abt it)?

Reply:
I think hermes does have it. They have tests for this method https://github.com/facebook/hermes/blob/main/test/hermes/array-flat-recursion.js

Reply:
No, it's facebook fauth, they stopped update hermes changelogs. Now only facebook devs can know what JS features is are supported in what react native version

Reply:
wait for new app update, we are going to migrate to new arch of React Native

Reply:
No, but could you write a guide to use this one, I have no idea to use it

Reply:
ye